Content.Server/Decals/DecalSystem.cs

```diff
@@ -0,0 +1,323 @@
+        {
+            if (!ChunkIndex.TryGetValue(gridId, out var values) || !values.TryGetValue(uid, out var indices))
+            {
+                return false;
+            }
+
+            var chunkCollection = ChunkCollection(gridId);
+            var decal = chunkCollection[indices][uid];
+            chunkCollection[indices][uid] = decal.WithColor(color);
+            DirtyChunk(gridId, indices);
+            return true;
+        }
+
+        public bool SetDecalId(GridId gridId, uint uid, string id)
+        {
+            if (!ChunkIndex.TryGetValue(gridId, out var values) || !values.TryGetValue(uid, out var indices))
+            {
+                return false;
+            }
+
+            if (!PrototypeManager.HasIndex<DecalPrototype>(id))
+                throw new ArgumentOutOfRangeException($"Tried to set decal id to invalid prototypeid: {id}");
+
+            var chunkCollection = ChunkCollection(gridId);
+            var decal = chunkCollection[indices][uid];
+            chunkCollection[indices][uid] = decal.WithId(id);
+            DirtyChunk(gridId, indices);
+            return true;
+        }
+
+        public bool SetDecalRotation(GridId gridId, uint uid, Angle angle)
+        {
+            if (!ChunkIndex.TryGetValue(gridId, out var values) || !values.TryGetValue(uid, out var indices))
+            {
+                return false;
+            }
+
+            var chunkCollection = ChunkCollection(gridId);
+            var decal = chunkCollection[indices][uid];
+            chunkCollection[indices][uid] = decal.WithRotation(angle);
+            DirtyChunk(gridId, indices);
+            return true;
+        }
+
+        public bool SetDecalZIndex(GridId gridId, uint uid, int zIndex)
+        {
+            if (!ChunkIndex.TryGetValue(gridId, out var values) || !values.TryGetValue(uid, out var indices))
+            {
+                return false;
+            }
+
+            var chunkCollection = ChunkCollection(gridId);
+            var decal = chunkCollection[indices][uid];
+            chunkCollection[indices][uid] = decal.WithZIndex(zIndex);
+            DirtyChunk(gridId, indices);
+            return true;
+        }
+
+        public bool SetDecalCleanable(GridId gridId, uint uid, bool cleanable)
+        {
+            if (!ChunkIndex.TryGetValue(gridId, out var values) || !values.TryGetValue(uid, out var indices))
+            {
+                return false;
+            }
+
+            var chunkCollection = ChunkCollection(gridId);
+            var decal = chunkCollection[indices][uid];
+            chunkCollection[indices][uid] = decal.WithCleanable(cleanable);
+            DirtyChunk(gridId, indices);
+            return true;
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            foreach (var playerSession in _playerManager.GetAllPlayers())
+            {
+                var chunks = GetChunksForSession(playerSession);
```

Comment:
Do the warning.
Also if you check for InGame here it avoids needing to allocate updatedChunks before realising they're not ingame.

Reply:
dang, totally missed that. what do you mean by "do the warning"?